### ISSUE TITLE
[feature] summon-workspace: don't switch monitors

### ISSUE THREAD

**Issue by raisjn** [nikitabobko/AeroSpace#603]
This feature is very cool! Aerospace is almost perfect for me now :-D 

Is it possible to add a parameter to `summon-workspace` so that when focusing to an active workspace on a separate monitor, it brings that workspace to the current monitor? This is how my xmonad is currently setup. 

Current behavior: the other monitor will become focused. If we are at [0,1*], it becomes [0*, 1]
Desired behavior: the two workspaces will be swapped. If it was [0,1*] before, it would be [1, 0*]. 

I have modified the source to accomplish this for myself, but would appreciate it being included in your source for future releases. Thank you! 

## Checklist

- [x] I've searched for duplicates. My search queries were (please don't forget to search in closed issues too):
  - xmonad
  - summon-workspace
- [x] I've read the guide https://nikitabobko.github.io/AeroSpace/guide
- [x] I've searched in the documentation (https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands). 

My search queries were:
  - summon-workspace

**Comment by nikitabobko** [comment]
Yes, I realized that AeroSpace behavior isn't identical to Xmonad after 0.15.0 was released. I agree that swapping the workspaces should be the default.

As an Xmonad user, can you please tell how is the appropriate operation called in Xmonad?

**Comment by raisjn** [comment]
> When you have multiple workspaces visible and mod-n to a different visible workspace, your current one swaps with the other one. We'll see how to change that below, if you don't like the swapping -- simply change 'greedyView' to 'view' in your workspace key bindings

from https://wiki.haskell.org/Xmonad/Frequently_asked_questions#Multi_head_and_workspaces_.28desktops.29

in short: swapping the workspaces (and keeping focus on current monitor) is called "greedyView" and is default. not swapping workspace + moving focus is known as "view"

**Comment by raisjn** [comment]
i'm going to daily drive this and try to implement + document the missing features from xmonad that i use frequently. a quick list so far (please don't feel the need to implement these, just putting this here for myself for now)

1. is there a keybinding for swapping the contents of the monitor. i suspect there may be a sequence of commands to accomplish it
2. a command to focus the next and previous windows (instead of directional only) with wrap around. 
3. using `layout accordion` doesn't seem to embed the current window back into the container, but `layout tiling` does. i typically bind one key for pushing a window back into the tiling order and have separate keys for switching between layouts. 
4. dragging a window should be able to pull it out of tiling mode (and into floating mode), then the key from 3. should be able to push it back into tiling mode

**Comment by nikitabobko** [comment]
> i'm going to daily drive this and try to implement + document the missing features from xmonad that i use frequently. 

That would be very welcome, thanks!

1. Swapping contenst of monitors == swapping workspaces, which is very similar to workspace renaming #245 
2. #248 
3. Please make sure to not confuse `tiling` and `tiles` layouts
4. I don't mind to have a separate config option like `windows-mouse-dragging = 'swap|float'`. It should be easy to implement. Feel free to create an issue or a PR. But I won't be working on that, I don't care about floating windows and mouse dragging enough.

**Comment by raisjn** [comment]
1. i implemented a rotate workspaces command for myself and put it on alt-tab. the command will move visible workspaces over by one monitor, but i could see it also have args to swap contents of specific monitors if people have more than one monitor. 
2. i'll look into this for myself. like ntrp i typically only have 2 - 3 windows on a workspace at most. 
3. ah, that is confusing - i see what happened - thank you
4. do you have a code pointer for this? (no rush, though)

---

i realized the diff i put up above is not exactly what i want. i've been using a different diff, but when i think about the behavior i want, it is a bit complicated. 

let's say we have monitors [1,2,3] which are unpinned and monitors [A, B, C] (forced to primary) and [X, Y, Z] (forced to secondary).

* summoning A,B,C,X,Y,Z should change it's pinned screen to that workspace. if we have [1, 2*] and we call `summon-workspace A`, it should become [A, 2]. if we call `summon-workspace Z`, it should become [1, Z*]. 
* a pinned workspace should not leave its pinned monitor when swapping workspace. this becomes tricky in the case of [1, Z*] and calling `summon-workspace 1`. what should happen here? we don't want to push Z to primary because it's pinned to secondary, but how to know what to place on primary monitor? i will think more on this.

**Comment by raisjn** [comment]
I'll close this and keep modifications in a private fork, thanks for the discussion and great tool!
